elmslie-core/src/main/java/vivid/money/elmslie/core/store/Store.kt

```diff
@@ -23,38 +24,23 @@ interface Store<Event, Effect, State> {
     fun accept(event: Event)
 
     /**
-     * Provides ability to subscribe to state changes.
+     * Returns the flow of [State]. Internally the store keeps the last emitted state value, so each
+     * new subscribers will get it.
      *
-     * State dispatching is restricted. Behavior contract:
-     * - The current state will be sent synchronously.
-     * - Every two subsequent invocation of [onStateChange] have not equal states.
-     * - States are **never** delivered on the main thread.
+     * Note that there will be no emission if a state isn't changed (it's [equals] method returned
+     * `true`.
      *
-     * @return [Disposable] For stopping [onStateChange] callback invocations.
+     * By default, [State] is collected in [Dispatchers.IO].
      */
-    fun states(onStateChange: (State) -> Unit): Disposable
+    fun states(): Flow<State>
 
     /**
-     * Provides ability to subscribe to effect emissions.
+     * Returns the flow of [Effect]. It's a _hot_ flow and values produced by it **don't cache**.
```

Comment:
Really? I think we can implement hot flow in many ways under the hood) of course the most wise will be using SharedFlow, but still. We only highlight that it's hot, in case if user wants to understand the pipeline